src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -375,45 +238,146 @@ public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      this.conf.getRestApiEndpoint(), tableName, false)
+      .setExcludeReferenceLink(true)
+      .setDisplayValue(SourceValueType.SHOW_DISPLAY_VALUE)
+      .setLimit(1);
+    RestAPIResponse apiResponse = null;
+    String accessToken = getAccessToken();
+    requestBuilder.setResponseHeaders(ServiceNowConstants.HEADER_NAME_TOTAL_COUNT);
+    requestBuilder.setAuthHeader(accessToken);
+    apiResponse = executeGet(requestBuilder.build());
+    if (!apiResponse.isSuccess()) {
+      throw new RuntimeException("Error : " + apiResponse);
+    }
+    return getRecordCountFromHeader(apiResponse);
+  }
+
+  /**
+   * Create a new record in the ServiceNow Table
+   *
+   * @param tableName ServiceNow Table name
+   * @param entity    Details of the Record to be created
+   * @description This function is being used in end-to-end (e2e) tests to fetch a record from the ServiceNow Table.
+   */
+  public String createRecord(String tableName, HttpEntity entity) throws IOException {
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      this.conf.getRestApiEndpoint(), tableName, false);
+    String systemID;
+    RestAPIResponse apiResponse = null;
     try {
       String accessToken = getAccessToken();
       requestBuilder.setAuthHeader(accessToken);
-      apiResponse = executeGet(requestBuilder.build());
+      requestBuilder.setAcceptHeader("application/json");
+      requestBuilder.setContentTypeHeader("application/json");
+      requestBuilder.setEntity(entity);
+      apiResponse = executePost(requestBuilder.build());
+
+      systemID = String.valueOf(getSystemId(apiResponse));
+
       if (!apiResponse.isSuccess()) {
         LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
-        collector.addFailure("Unable to fetch schema for table " + tableName, null).
-          withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAME);
-        return null;
-      }
-      List<SchemaResponse> result = parseSchemaResponse(apiResponse.getResponseBody());
-      List<ServiceNowColumn> columns = new ArrayList<>();
-
-      if (result != null && !result.isEmpty()) {
-        for (SchemaResponse field : result) {
-          columns.add(new ServiceNowColumn(field.getName(), field.getInternalType()));
-        }
+      } else {
+        LOG.info(apiResponse.getResponseBody());
       }
-      return SchemaBuilder.constructSchema(tableName, columns);
-    } catch (OAuthSystemException | OAuthProblemException e) {
-      LOG.error("Error in connection - {}", e);
-      collector.addFailure("Connection failed. Unable to fetch schema for table " + tableName, null);
+    } catch (OAuthSystemException | OAuthProblemException | UnsupportedEncodingException e) {
+      LOG.error("Error in creating a new record", e);
+      throw new RuntimeException("Error in creating a new record");
```

Comment:
when wrapping exceptions, the cause must always be specified, otherwise the call stack is lost.

```
throw new RuntimeException(..., e);
```

that being said, it's normally not a good idea to rethrow checked exceptions as runtime exceptions in code that we completely control. We should either be propagating the exception up instead of catching and rethrowing (normally better), or we should wrap it in an IOException instead of RuntimeException.

Reply:
Changed to IOException and specified the casue. e63c214cf864e664ca24072f21c719c1a777b9c4